qadence/register.py

```diff
@@ -50,22 +52,37 @@ def __init__(self, support: nx.Graph | int):
         """
         self.graph = support if isinstance(support, nx.Graph) else alltoall_graph(support)
 
+        if scale != 1.0:
+            _scale_node_positions(self.graph, scale)
+
+        # Auxiliary complete graph
+        support = self.graph.nodes
+        all_edges = list(filter(lambda x: x[0] < x[1], product(support, support)))
+        self.complete_graph = nx.Graph()
+        self.complete_graph.add_nodes_from(support)
+        self.complete_graph.add_edges_from(all_edges)
```

Comment:
I use it in this MR: https://github.com/pasqal-io/qadence/pull/176, but I figured I would put the changes here since they relate to the register. But actually there's something else I need to add so that `hamiltonian_factory` properly works with it.

Basically `hamiltonian_factory(register, ...)` will create a hamiltonian based on the edges of the register. But if I want for example a hamiltonian dependent on the distance between the qubits, independently of the edges in the graph, that's what this `complete_graph` is used for.

Reply:
awesome that sounds great 👍